grakn-graph/src/main/java/ai/grakn/graph/internal/RelationImpl.java

```diff
@@ -63,15 +63,16 @@
      */
     Optional<RelationReified> reified(){
         if(!relationStructure.isReified()) return Optional.empty();
-        return Optional.of(relationStructure.reified());
+        return Optional.of(relationStructure.reify());
     }
 
     /**
      * Reifys and returns the {@link RelationReified}
      */
     RelationReified reify(){
-        if(relationStructure.isReified()) return relationStructure.reified();
-        throw new UnsupportedOperationException("Reification is not yet supported");
+        if(relationStructure.isReified()) return relationStructure.reify();
+        relationStructure = relationStructure.reify();
+        return relationStructure.reify();
```

Comment:
Last line should just be `return relationStructure;`

We can also just remove the check in the first line, but I guess it skips an assignment? I don't know if it's worth the cost of the extra function call though.